src/output-json-file.c

```diff
@@ -78,6 +78,42 @@ typedef struct JsonFileLogThread_ {
     MemBuffer *buffer;
 } JsonFileLogThread;
 
+/* for hashing truncated files... */
+#ifdef HAVE_NSS
+static void FileWriteJsonHash( const File *ff, json_t *fjs )
```

Comment:
I think it makes sense to also use this in the default FILE_STATE_CLOSED case

Reply:
But the default state seems to be UNKNOWN, not FILE_STATE_CLOSED, that's another case different from the default one and the hash its already calculated within so I suppose you're referring to the UNKNOWN default case.

Sounds right to you or am I wrong?

And what about FILE_STATE_ERROR? Should I add the "FileWriteJsonHash()" there?

Reply:
I meant call this util func for case FILE_STATE_CLOSED: as well. Now the code is duplicated between FileWriteJsonRecord and FileWriteJsonHash

Reply:
Ah! Ok... :)